lib/super_digit.rb

```diff
@@ -1,15 +1,18 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - ? O(n)
+# Space Complexity - ? O(n^2)
 def super_digit(n)
-   
+  return n if n < 10
+  super_digit(n.digits.sum)
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - ?O(n + k)
+# Space Complexity - ? O(n)
 def refined_super_digit(n, k)
```

Comment:
Your time complexity is going to be O(n * k) look below that you're increasing the number of digits by multiplying them by k.